ts/Grid/Core/Pagination/Pagination.ts

```diff
@@ -125,21 +177,56 @@ class Pagination {
      */
     constructor(
         grid: Grid,
-        options: PaginationOptions
+        options: PaginationOptions & Pagination.PaginationState
     ) {
         this.grid = grid;
-        this.options = options ?? Pagination.defaultOptions;
-        this.itemsPerPage = this.options.itemsPerPage ||
-            Pagination.defaultOptions.itemsPerPage as number;
-    }
+        this.options = merge(Pagination.defaultOptions, options);
+
+        this.pageSizeOptions =
+            this.options.controls.pageSizeSelector.options as number[];
+
+        this.currentPageSize =
+            options.currentPageSize ||
+            this.options.pageSize ||
+            this.pageSizeOptions[0];
 
+        // Lang pack
+        this.lang = merge(
+            Defaults.defaultOptions.pagination,
+            this.grid.options?.lang?.pagination
+        );
+
+        // Set state
+        if (options.currentPage) {
+            this.currentPage = options.currentPage;
+        }
+        if (options.totalItems) {
+            this.totalItems = options.totalItems;
+        }
+        if (options.totalPages) {
+            this.totalPages = options.totalPages;
+        }
+    }
 
     /* *
     *
     *  Methods
     *
     * */
 
+    /**
+     * Format text with placeholders.
+     *
+     * @param template The text template with placeholders
+     * @param values Object containing values to replace placeholders
+     * @returns Formatted text
+     */
+    private formatText(template: string, values: Record<string, string | number>): string {
```

Comment:
What this code does is a bit unclear to me at first sight (but maybe it's just me). Can we change its description or move the `/\{(\w+)\}/g` into a variable? Also, this seems more like a util thing. Have you thought about extracting it from the class?